custom_components/tahoma/sensor.py

```diff
@@ -1,116 +1,338 @@
-CORE_WATER_CONSUMPTION_STATE = "core:WaterConsumptionState"
-CORE_WINDSPEED_STATE = "core:WindSpeedState"
-
-
-DEVICE_CLASS_SUN_ENERGY = "sun_energy"
-DEVICE_CLASS_WIND_SPEED = "wind_speed"
-
-ICON_MOLECULE_CO = "mdi:molecule-co"
-ICON_MOLECULE_CO2 = "mdi:molecule-co2"
-ICON_SOLAR_POWER = "mdi:solar-power"
-ICON_WEATHER_WINDY = "mdi:weather-windy"
-
-UNIT_LX = "lx"
-
-TAHOMA_SENSOR_DEVICE_CLASSES = {
-    "CO2Sensor": DEVICE_CLASS_CO2,
-    "COSensor": DEVICE_CLASS_CO,
-    "ElectricitySensor": DEVICE_CLASS_POWER,
-    "HumiditySensor": DEVICE_CLASS_HUMIDITY,
-    "LightSensor": DEVICE_CLASS_ILLUMINANCE,
-    "RelativeHumiditySensor": DEVICE_CLASS_HUMIDITY,
-    "SunSensor": DEVICE_CLASS_SUN_ENERGY,
-    "TemperatureSensor": DEVICE_CLASS_TEMPERATURE,
-    "WindSensor": DEVICE_CLASS_WIND_SPEED,
-}
-# From https://www.tahomalink.com/enduser-mobile-web/steer-html5-client/tahoma/bootstrap.js
-UNITS = {
-    "core:TemperatureInCelcius": TEMP_CELSIUS,
-    "core:TemperatureInCelsius": TEMP_CELSIUS,
-    "core:TemperatureInKelvin": TEMP_KELVIN,
-    "core:TemperatureInFahrenheit": TEMP_FAHRENHEIT,
-    "core:LuminanceInLux": UNIT_LX,
-    "core:ElectricCurrentInAmpere": ELECTRIC_CURRENT_AMPERE,
-    "core:VoltageInVolt": ELECTRIC_POTENTIAL_VOLT,
-    "core:ElectricalEnergyInWh": ENERGY_WATT_HOUR,
-    "core:ElectricalEnergyInKWh": ENERGY_KILO_WATT_HOUR,
-    "core:ElectricalEnergyInMWh": f"M{ENERGY_WATT_HOUR}",
-    "core:ElectricalPowerInW": POWER_WATT,
-    "core:ElectricalPowerInKW": POWER_KILO_WATT,
-    "core:ElectricalPowerInMW": f"M{POWER_WATT}",
-    "core:FlowInMeterCubePerHour": VOLUME_CUBIC_METERS,
-    "core:LinearSpeedInMeterPerSecond": SPEED_METERS_PER_SECOND,
-    "core:RelativeValueInPercentage": PERCENTAGE,
-    "core:VolumeInCubicMeter": VOLUME_CUBIC_METERS,
-    "core:VolumeInLiter": VOLUME_LITERS,
-    "core:FossilEnergyInWh": ENERGY_WATT_HOUR,
-    "core:FossilEnergyInKWh": ENERGY_KILO_WATT_HOUR,
-    "core:FossilEnergyInMWh": f"M{ENERGY_WATT_HOUR}",
-    "meters_seconds": SPEED_METERS_PER_SECOND,
-}
-
-UNITS_BY_DEVICE_CLASS = {
-    DEVICE_CLASS_CO2: CONCENTRATION_PARTS_PER_MILLION,
-    DEVICE_CLASS_CO: CONCENTRATION_PARTS_PER_MILLION,
-}
+
+@dataclass
+class OverkizSensorDescription(SensorEntityDescription):
+    """Class to describe a Overkiz sensor."""
+
+    value: Callable[[Any], Any] | None = lambda val: val
+
+
+SENSOR_DESCRIPTIONS = [
+    OverkizSensorDescription(
+        key="core:BatteryLevelState",
+        name="Battery Level",
+        unit_of_measurement=PERCENTAGE,
+        device_class=sensor.DEVICE_CLASS_BATTERY,
+        state_class=STATE_CLASS_MEASUREMENT,
+    ),
+    OverkizSensorDescription(
+        key="core:BatteryState",
+        name="Battery",
+        device_class=sensor.DEVICE_CLASS_BATTERY,
+        value=lambda value: str(value).capitalize(),
+    ),
+    OverkizSensorDescription(
+        key="core:RSSILevelState",
+        name="RSSI Level",
+        value=lambda value: round(value),
```

Comment:
Why round the value for every sensor? The raw data is not correct?

Reply:
Tbh, I don't recall why I added this. There are quite some states that give very precise measures, thus I chose to round it. 

Should I remove all the rounds and add them back based on user reports? Or should we keep it safe and just keep them?

Reply:
I would suggest to keep the raw value. And see if user complains.

Reply:
Done! I just added it for RSSI, since I can test this one and it always return .0, thus rounding will make it cleaner.